### PR TITLE
Wrap four new KtLint rules

### PR DESCRIPTION
**Pull request by arturbosch**

### CODE REVIEW

**Comment by arturbosch** [comment]
Thanks for the reminder, I even forgot to put them into the provider which brings me to the idea to `generate` the providers! (will open an issue)
Also the annotation rule allows two annotations on the same line (when no other source tokens are present), need to ask them if this is intentional. Nothing we can do now about it :).

**Comment by arturbosch** [comment]
Seems like a KtLint rule bug.

**Comment by 3flex** [comment]
Before enabling new ktlint rules, check if it's enabled upstream at https://github.com/shyiko/ktlint/blob/master/ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/StandardRuleSetProvider.kt

detekt can still wrap them but should disable by default in `default-detekt-config.yml`

AnnotationRule is disabled in ktlint so should probably not expect it to work properly at the moment.

**Comment by arturbosch** [comment]
Thanks for the link and the suggestion, will do it.

File: detekt-cli/src/main/resources/default-detekt-config.yml [hunk 1]
```diff
@@ -277,6 +279,8 @@ formatting:
   SpacingAroundCurly:
     active: true
     autoCorrect: true
+  SpacingAroundDot:
+    active: false
```
Comment (3flex) [inline detekt-cli/src/main/resources/default-detekt-config.yml]: This one could be enabled? It's enabled by ktlint. Of these two rules only `AnnotationRule` is disabled by default in ktlint, but this one should be OK to turn on like the rest of the formatting rules.

Comment (arturbosch) [reply detekt-cli/src/main/resources/default-detekt-config.yml]: It is actually a deal breaker, see https://github.com/shyiko/ktlint/issues/344 I opened just now ... :(

**Comment by arturbosch** [comment]
Also the new rule `SpacingAroundDot` interferes with `ChainWrapping`:
```
❯ diff /tmp/diff1 /tmp/diff2
3c3,5
<     val anchor = owner.firstChild!!.siblings(forward = true).dropWhile { it is PsiComment || it is PsiWhiteSpace }
---
>     val anchor = owner.firstChild!!
>         .siblings(forward = true)
>         .dropWhile { it is PsiComment || it is PsiWhiteSpace }

```

File: detekt-cli/src/main/resources/default-detekt-config.yml (hunk 1 above)
Comment (3flex) [reply detekt-cli/src/main/resources/default-detekt-config.yml]: Fix was released in klint 0.31.0

**Comment by 3flex** [comment]
There's a new rule in 0.33: SpacingAroundUnaryOperatorsRule

Also MultiLineIfElseRule is not wrapped.

There's also a new indentation rule in the experimental ruleset which may be worth wrapping also (ImportOrderingRule from experimental ruleset is already wrapped).

**Comment by arturbosch** [comment]
So this wraps four new rules. The codfactor issues are test resources. Need to investigate how to suppress them ^^.

**Comment by arturbosch** [comment]
CI fails due to https://github.com/pinterest/ktlint/issues/490
Should we disable the `Annotation` rule and get this merged?

**Comment by schalkms** [comment]
@arturbosch yes

**Comment by arturbosch** [comment]
Well I guess KtLint formats the file differently on java8 and java9+ ...

**Comment by schalkms** [comment]
Probably a bug?

**Pull request merged**